### PR TITLE
TS-89-refactor-Teachers-Page

### LINKED ISSUE #89: Refactor: TeacherPage and divide it into smaller components
Refactor TeacherPage and all components which belong to it.

Acceptance Criteria:
The TeacherPage is divided into smaller components and  its services are rewritten to sagas.

Sub Tasks \ Bugs:
 
- [ ] divide TeacherPage into smaller components ;

- [ ]  rewrite services to sagas.

- [ ] create search by departments

- [ ] write new functionality about links

Estimate: 4|5|6 days

### PR DESCRIPTION
**Pull request by Yashchemskyi-Oleksandr**

### CODE REVIEW

File: frontend/src/components/TeachersPage/AddTeacherForm/AddTeacherForm.js [hunk 1]
```diff
@@ -126,9 +125,9 @@ let AddTeacher = (props) => {
                         getDepartmentByIdService(target.value);
                     }}
                 >
-                    <option />
+                    <option className="option" />
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/TeachersPage/AddTeacherForm/AddTeacherForm.js]: Could you use more unique name?

File: frontend/src/components/TeachersPage/TeachersList/TeachersList.scss [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    box-shadow: 0 2px 8px #00000042;
+
+    .teacher-card-name {
+        height: 3.5em;
+        margin: 0.3em 0 1em;
+        font-size: 1.3rem;
+        text-overflow: ellipsis;
+        overflow: hidden;
+
+        @media (max-width: 400px) {
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/TeachersPage/TeachersList/TeachersList.scss]: add constants

File: frontend/src/components/TeachersPage/TeachersList/TeachersList.scss [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+.teacher-card {
+    position: relative;
+    height: 14.5em;
+    background-color: #fff;
+    width: 15em;
+    display: inline-block;
+    margin: 10px;
+    padding: 1rem;
+    border-radius: 5px;
+    box-shadow: 0 2px 8px #00000042;
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/TeachersPage/TeachersList/TeachersList.scss]: add constants to color

File: frontend/src/components/TeachersPage/TeachersList/TeachersList.scss [hunk 4]
```diff
@@ -0,0 +1,119 @@
+@import '../../../styles/colors.scss';
+@import '../../../styles/fonts.scss';
+@import '../../../styles/variables.scss';
+
+.teacher-card {
+    position: relative;
+    height: 14.5em;
+    background-color: #fff;
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/TeachersPage/TeachersList/TeachersList.scss]: use one style of color

File: frontend/src/components/TeachersPage/TeachersPage.js [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+        setSelected([]);
+    };
+
+    const isChosenSelection = () => {
+        return selected.length !== 0;
+    };
+
+    const parseDefaultSemester = () => {
+        return {
+            id: defaultSemester.id,
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/TeachersPage/TeachersPage.js]: Could you destructuring  defaultSemester?

File: frontend/src/constants/formats.js [hunk 6]
```diff
@@ -3,3 +3,7 @@ export const dateFormat = 'DD/MM/YYYY';
 export const timeFormat = 'HH:mm';
 
 export const hourFormat = 'h';
+
+export const MAX_LENGTH = 50;
+
+export const HTTP = 'http';
```
Comment (ViktoriiaKosovan) [inline frontend/src/constants/formats.js]: There are date constants, move MAX_LENGTH, 'http' to other file

File: frontend/src/containers/MergeRolePage/MergeRolePage.js [hunk 7]
```diff
@@ -39,10 +40,10 @@ const MergeRolePage = (props) => {
 
     const classes = useStyles();
 
-    const { teachers } = props;
+    const { teachers, gettTeachersWithoutAccount } = props;
```
Comment (ViktoriiaKosovan) [inline frontend/src/containers/MergeRolePage/MergeRolePage.js]: change gett to get

**Review by ViktoriiaKosovan — APPROVED** [review]

File: frontend/src/components/TeachersPage/TeachersPage.js [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+        });
+        const semesterId = selectedSemester === '' && defaultSemester.id;
+        const { language } = i18n;
+        const data = { semesterId, teachersId, language };
+        sendTeacherSchedule(data);
+        setSelected([]);
+    };
+
+    const isChosenSelection = () => {
+        return selected.length !== 0;
```
Comment (HiroFive) [inline frontend/src/components/TeachersPage/TeachersPage.js]: use isEmpty from `lodash`

**Review by HiroFive — APPROVED** [review]

File: frontend/src/components/TeachersPage/AddTeacherForm/AddTeacherForm.js (hunk 1 above)
Comment (Yashchemskyi-Oleksandr) [reply frontend/src/components/TeachersPage/AddTeacherForm/AddTeacherForm.js]: done

File: frontend/src/components/TeachersPage/TeachersList/TeachersList.scss (hunk 2 above)
Comment (Yashchemskyi-Oleksandr) [reply frontend/src/components/TeachersPage/TeachersList/TeachersList.scss]: done

File: frontend/src/components/TeachersPage/TeachersList/TeachersList.scss (hunk 3 above)
Comment (Yashchemskyi-Oleksandr) [reply frontend/src/components/TeachersPage/TeachersList/TeachersList.scss]: done

File: frontend/src/components/TeachersPage/TeachersList/TeachersList.scss (hunk 4 above)
Comment (Yashchemskyi-Oleksandr) [reply frontend/src/components/TeachersPage/TeachersList/TeachersList.scss]: done

File: frontend/src/components/TeachersPage/TeachersPage.js (hunk 5 above)
Comment (Yashchemskyi-Oleksandr) [reply frontend/src/components/TeachersPage/TeachersPage.js]: done

File: frontend/src/constants/formats.js (hunk 6 above)
Comment (Yashchemskyi-Oleksandr) [reply frontend/src/constants/formats.js]: done

File: frontend/src/containers/MergeRolePage/MergeRolePage.js (hunk 7 above)
Comment (Yashchemskyi-Oleksandr) [reply frontend/src/containers/MergeRolePage/MergeRolePage.js]: done

File: frontend/src/components/TeachersPage/TeachersPage.js (hunk 8 above)
Comment (Yashchemskyi-Oleksandr) [reply frontend/src/components/TeachersPage/TeachersPage.js]: done

**Review by Niknikson — APPROVED** [review]

**Pull request closed without merging**